src/Compilers/Core/Portable/DiagnosticAnalyzer/SuppressMessageAttributeState.cs

```diff
@@ -244,11 +257,14 @@ private void DecodeGlobalSuppressMessageAttributes()
 
                 Interlocked.CompareExchange(ref _lazyGlobalSuppressions, suppressions, null);
             }
+            return _lazyGlobalSuppressions;
         }
 
         private ImmutableDictionary<string, SuppressMessageInfo> DecodeLocalSuppressMessageAttributes(ISymbol symbol)
         {
-            var attributes = symbol.GetAttributes().Where(a => a.AttributeClass == this.SuppressMessageAttribute);
+            var attributes = symbol.GetAttributes().Where(a =>
+                a.AttributeClass == this.SuppressMessageAttribute
+                || a.AttributeClass == UnconditionalSuppressMessageAttribute);
```

Comment:
It looks like this condition is used in at least two places, consider extracting and reusing a helper.